### PR TITLE
Misc modifications, mostly PDep related

### PR DESCRIPTION
**Pull request by alongd**
Several enhancements related to PDep:
-  Don't look in METHOD_MAP a second time: When writing an RMG input file, pdep['method'] is being assigned a value according to the respective acronym. After assigning the value, we should not look again in the METHOD_MAP dict.
- Only compute thermo for an SA observable if it requires refinement
- Check that core_tolerance is not a float before determining its length
- BugFix: Modify a copy of pdep in writer
- Check multiplicity as well when performing isomorphism check (test added)

### CODE REVIEW

**Comment by alongd** [comment]
@kspieks, this branch should be ready for review now, I'll open a parallel PR on RMG, we might want to merge the RMG one first.

**Comment by alongd** [comment]
@kspieks, this PR was updated and should be ready for review

**Review by kspieks — APPROVED** [review]
Thanks for the updates! I only had one question, mostly just to learn. Otherwise, the commits look good.

File: t3/main.py [hunk 1]
```diff
@@ -1334,6 +1334,6 @@ def get_species_label_by_structure(adj: str,
     """
     new_spc = Species().from_adjacency_list(adj)
     for spc in species_list:
-        if spc.is_isomorphic(new_spc):
+        if spc.is_isomorphic(new_spc) and spc.molecule[0].multiplicity == new_spc.molecule[0].multiplicity:
```
[the remark this answers is not in the document]
Comment (alongd) [reply t3/main.py:1338]: Good point, you're right, it's not needed. Was added to overcome the bug we had when we previously used SMILES, not needed anymore.

**Comment by alongd** [comment]
I removed the multiplicity checking commit, should be ready to go now. Thanks :)

File: t3/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kspieks) [reply t3/main.py]: Sounds good. I'll merge now. Thanks for the PR!

**Review by kspieks — APPROVED** [review]

**Pull request merged**